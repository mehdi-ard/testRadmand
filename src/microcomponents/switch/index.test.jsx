import React from 'react';
import renderer from 'react-test-renderer';
import Switch from '.';

it('renders default', () => {
  expect(renderer.create(<Switch />).toJSON()).toMatchSnapshot();
});
it('renders with label', () => {
  expect(
    renderer.create(<Switch title="default" />).toJSON()
  ).toMatchSnapshot();
});

it('renders size', () => {
  expect(
    renderer.create(<Switch size="small" title="default" />).toJSON()
  ).toMatchSnapshot();
});

it('renders with text', () => {
  expect(
    renderer
      .create(<Switch checkedChildren="ON" unCheckedChildren="OFF" />)
      .toJSON()
  ).toMatchSnapshot();
});
