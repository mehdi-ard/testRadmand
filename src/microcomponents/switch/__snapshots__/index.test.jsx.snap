// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default 1`] = `
Array [
  <span
    className="ant-typography"
    style={
      Object {
        "WebkitLineClamp": null,
      }
    }
  >
    
  </span>,
  <button
    aria-checked={false}
    className="ant-switch"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="switch"
    type="button"
  >
    <div
      className="ant-switch-handle"
    />
    <span
      className="ant-switch-inner"
    />
  </button>,
]
`;

exports[`renders size 1`] = `
Array [
  <span
    className="ant-typography"
    style={
      Object {
        "WebkitLineClamp": null,
      }
    }
  >
    default
  </span>,
  <button
    aria-checked={false}
    className="ant-switch ant-switch-small"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="switch"
    type="button"
  >
    <div
      className="ant-switch-handle"
    />
    <span
      className="ant-switch-inner"
    />
  </button>,
]
`;

exports[`renders with label 1`] = `
Array [
  <span
    className="ant-typography"
    style={
      Object {
        "WebkitLineClamp": null,
      }
    }
  >
    default
  </span>,
  <button
    aria-checked={false}
    className="ant-switch"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="switch"
    type="button"
  >
    <div
      className="ant-switch-handle"
    />
    <span
      className="ant-switch-inner"
    />
  </button>,
]
`;

exports[`renders with text 1`] = `
Array [
  <span
    className="ant-typography"
    style={
      Object {
        "WebkitLineClamp": null,
      }
    }
  >
    
  </span>,
  <button
    aria-checked={false}
    className="ant-switch"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="switch"
    type="button"
  >
    <div
      className="ant-switch-handle"
    />
    <span
      className="ant-switch-inner"
    >
      OFF
    </span>
  </button>,
]
`;
